build-system/dep-check-config.js

```diff
@@ -246,6 +246,12 @@ exports.rules = [
       // Ads needs iframe transports
       'extensions/amp-ad-exit/0.1/config.js->extensions/amp-analytics/0.1/iframe-transport-vendors.js',
 
+      // Facebook components
+      'extensions/amp-facebook/0.1/amp-facebook.js->extensions/amp-facebook/facebook-loader.js',
```

Comment:
I think these will require build-system changes, too. @rsimha?

Reply:
Not sure about that. In fact, I'm surprised this needs to be written a rule, since the dependency is within the same component.

Reply:
Components start at the version directory, not the generic component directory.

Reply:
@sparhami @jridgewell Bumping this thread due to these (currently silenced) type checking warnings that I'd like to fix.

Any reason `facebook-loader.js` needs to live in the generic component directory and not the version directory?

```js
Type checking failed:
extensions/amp-facebook-comments/0.1/amp-facebook-comments.js:17: ERROR - [JSC_JS_MODULE_LOAD_WARNING] Failed to load module "../../amp-facebook/facebook-loader"
import {createLoaderLogo} from '../../amp-facebook/facebook-loader';
^

extensions/amp-facebook-page/0.1/amp-facebook-page.js:17: ERROR - [JSC_JS_MODULE_LOAD_WARNING] Failed to load module "../../amp-facebook/facebook-loader"
import {createLoaderLogo} from '../../amp-facebook/facebook-loader';
^

extensions/amp-facebook/0.1/amp-facebook.js:17: ERROR - [JSC_JS_MODULE_LOAD_WARNING] Failed to load module "../facebook-loader"
import {createLoaderLogo} from '../facebook-loader';
^
```

Reply:
I poked through the build-system code, and it turns out that all extension code is expected to live in a directory of the format `amp-foo/0.x/` (under a given version). Sent out #23897 to move the file into the version directory.